### PR TITLE
Audit log the User-Agent

### LINKED ISSUE #3769: Admin audit app should log user-agent
Hi,

For audit logging, there are use cases where knowing the user-agent is useful. Therefore an enhancement would be great which enables the admin audit app to log the user-agent. Although the webserver is capable of logging it in the access log, it is a hard job to merge the audit and access log entries. Thank you.

### PR DESCRIPTION
**Pull request by GitHubUser4234**
Fixes #3769

### CODE REVIEW

**Review by LukasReschke — CHANGES_REQUESTED** [review]
I think that's an useful addition, so I'd go a little bit more crazy here and would suggest to even add this to  `\OC\Log\File::write`. Because that seems to come handy for debugging issues as well.

See for example the `user` or `method` there. Also please read from `$request->getHeader()` there :)

**Comment by LukasReschke** [comment]
But awesome stuff! 🚀

**Comment by GitHubUser4234** [comment]
> I think that's an useful addition, so I'd go a little bit more crazy here and would suggest to even add this to \OC\Log\File::write. Because that seems to come handy for debugging issues as well.
> 
> See for example the user or method there. Also please read from $request->getHeader() there :)

Superb, that gives a great boost to the usefulness of this thing :D    I just changed it accordingly.

File: lib/private/Log/File.php [hunk 1]
```diff
@@ -101,6 +101,7 @@ public static function write($app, $message, $level) {
 		} else {
 			$user = '--';
 		}
+		$userAgent = ($request->getHeader('User-Agent') !== '') ? $request->getHeader('User-Agent') : '--';
```
Comment (nickvergessen) [inline lib/private/Log/File.php]: For more fanciness 😎 
```php
$userAgent = $request->getHeader('User-Agent') ?: '--';
```

**Comment by GitHubUser4234** [comment]
> For more fanciness 😎
> 
> $userAgent = $request->getHeader('User-Agent') ?: '--';

Thanks, applied. This looks much better as it also handles the null case 👍

**Comment by nickvergessen** [comment]
It also handles `'0'` but thats no valid user agent either.

**Review by rullzer — APPROVED** [review]

**Review by LukasReschke — APPROVED** [review]

**Comment by LukasReschke** [comment]
Thanks, @GitHubUser4234 😄

**Pull request merged**